### PR TITLE
Moving common classes to azure-core-amqp from azure-core

### PR DESCRIPTION
**Pull request by conniey**
* Adding azure-core-amqp
* Moves it out of azure-core project since it only deals with AMQP

### CODE REVIEW

File: core/azure-core-amqp/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+If you would like to become an active contributor to this project please follow the instructions provided in [Microsoft Azure Projects Contribution Guidelines](http://azure.github.io/guidelines.html).
+
+1. Fork it
+1. Create your feature branch (`git checkout -b my-new-feature`)
+1. Commit your changes (`git commit -am 'Add some feature'`)
+1. Push to the branch (`git push origin my-new-feature`)
+1. Create new Pull Request
+
+<!-- Links -->
+[jdk]: https://docs.microsoft.com/en-us/java/azure/java-supported-jdk-runtime?view=azure-java-stable
```
Comment (JonathanGiles) [inline core/azure-core-amqp/README.md]: This link 404's. I would just remove the link entirely.

Comment (conniey) [reply core/azure-core-amqp/README.md]: I copied it from the application configuration README. Lol. Should I remove that too?
How about this link? 🤔 
https://docs.microsoft.com/en-us/java/azure/java-sdk-azure-get-started?view=azure-java-stable

Comment (JonathanGiles) [reply core/azure-core-amqp/README.md]: I would sooner just remove the link from all places. Leave it as an exercise to the reader to find the JDK they want to use.

Comment (conniey) [reply core/azure-core-amqp/README.md]: Gotcha. :) Sounds good. Removed.

**Pull request merged**